wp-content/civi-extensions/goonjcustom/Civi/CollectionCampHelper.php

```diff
@@ -0,0 +1,131 @@
+<?php
+
+namespace Civi;
+
+use Civi\Core\Service\AutoSubscriber;
+use CRM_Pets_ExtensionUtil as E;
+
+class CollectionCampHelper extends AutoSubscriber {
+
+	public static function getSubscribedEvents() {
+		return [
+			'&hook_civicrm_post' => 'generateCollectionCampCode',
+		];
+	}
+
+	/**
+	 * This hook is called after a db write on entities.
+	 *
+	 * @param string $op
+	 *   The type of operation being performed.
+	 * @param string $objectName
+	 *   The name of the object.
+	 * @param int $objectId
+	 *   The unique identifier for the object.
+	 * @param object $objectRef
+	 *   The reference to the object.
+	 */
+	public static function generateCollectionCampCode(string $op, string $objectName, int $objectId, &$objectRef) {
+		// Check if the object name is 'AfformSubmission'
+		if ($objectName !== 'AfformSubmission') {
+			return;
+		}
+
+		// Extract the 'data' field
+		$data = $objectRef->data;
+		$decodedData = json_decode($data, true);
+
+		// Access the subtype
+		$subtype = $decodedData['Eck_Collection_Camp1'][0]['fields']['subtype'] ?? null;
+
+		// Check if the subtype is 4 or 5
+		if ($subtype != 4 && $subtype != 5) {
+			return;
+		}
+
+		// Access the id within the decoded data
+		if (!isset($decodedData['Eck_Collection_Camp1'][0]['fields']['id'])) {
+			return;
+		}
+
+		$campId = $decodedData['Eck_Collection_Camp1'][0]['fields']['id'];
+
+		// Fetch the collection camp details
+		$collectionCamps = \Civi\Api4\EckEntity::get('Collection_Camp', FALSE)
+			->addWhere('id', '=', $campId)
+			->setLimit(1)
+			->execute();
+
+		if (empty($collectionCamps)) {
+			return;
+		}
+
+		$collectionCampsCreatedDate = $collectionCamps->first()['created_date'] ?? null;
+
+		// Get the year
+		$year = date('Y', strtotime($collectionCampsCreatedDate));
+
+		// Fetch the state ID from the collection camp intent details
+		$stateId = $decodedData['Eck_Collection_Camp1'][0]['fields']['Collection_Camp_Intent_Details.State'] ?? null;
+		if ($subtype == 5) {
+			$stateId = $decodedData['Eck_Collection_Camp1'][0]['fields']['Dropping_Centre.State'] ?? null;
+		}
+
+		if (!$stateId) {
+			return;
+		}
+
+		// Fetch the state abbreviation
+		$stateProvinces = \Civi\Api4\StateProvince::get(TRUE)
+			->addWhere('id', '=', $stateId)
+			->setLimit(1)
+			->execute();
+
+		if (empty($stateProvinces)) {
+			return;
+		}
+
+		$stateAbbreviation = $stateProvinces->first()['abbreviation'] ?? null;
+
+		if (!$stateAbbreviation) {
+			return;
+		}
+
+		// Fetch the Goonj-specific state code
+		$goonjStateCodePath = ABSPATH . 'wp-content/civi-extensions/goonjcustom/config/constants.php';
+		$goonjStateCode = include $goonjStateCodePath;
+		$stateCode = $goonjStateCode[$stateAbbreviation] ?? 'UNKNOWN';
+
+		// Get the current event title
+		$currentTitle = $decodedData['Eck_Collection_Camp1'][0]['fields']['title'] ?? 'Collection Camp';
+
+		// Fetch the event code
+		$eventCodePath = ABSPATH . 'wp-content/civi-extensions/goonjcustom/config/eventCode.php';
```

Comment:
We should find a different way to create this path. This path is WordPress specific however we are writing code in the Civi extension which cannot have CMS specific code.

Reply:
sure we can do that